static/js/reports/report.js

```diff
@@ -0,0 +1,1051 @@
+    var 
+          ONE_MIN_IN_MS = 60000
+        , SIX_MINS_IN_MS =  360000
+        , FORMAT_TIME_12 = '%I'
+        , FORMAT_TIME_24 = '%H'
+		;
+
+	var 
+		  SCALE_LINEAR = 0
+		, SCALE_LOG = 1;
+	
+		
+	var categories = [];
+	var foodlist = [];
+	
+    var padding = { top: 15, right: 15, bottom: 30, left: 35 };
+		
+     function getTimeFormat() {
+        var timeFormat = FORMAT_TIME_12;
+        if (browserSettings.timeFormat) {
+            if (browserSettings.timeFormat == '24') {
+                timeFormat = FORMAT_TIME_24;
+            }
+        }
+        return timeFormat;
+    }
+
+    // lixgbg: Convert mg/dL BG value to metric mmol
+    function scaleBg(bg) {
+        if (browserSettings.units == 'mmol') {
+            return (Math.round((bg / 18) * 10) / 10).toFixed(1);
+        } else {
+            return bg;
+        }
+    }
+
+    function rawIsigToRawBg(entry, cal) {
+
+      var raw = 0
+        , unfiltered = parseInt(entry.unfiltered) || 0
+        , filtered = parseInt(entry.filtered) || 0
+        , sgv = entry.y
+        , scale = parseFloat(cal.scale) || 0
+        , intercept = parseFloat(cal.intercept) || 0
+        , slope = parseFloat(cal.slope) || 0;
+
+
+        if (slope == 0 || unfiltered == 0 || scale == 0) {
+          raw = 0;
+        } else if (filtered == 0 || sgv < 40) {
+            raw = scale * (unfiltered - intercept) / slope;
+        } else {
+            var ratio = scale * (filtered - intercept) / slope / sgv;
+            raw = scale * ( unfiltered - intercept) / slope / ratio;
+        }
+
+        return Math.round(raw);
+    }
+
+    function drawChart(day,data,options) {
+       var   tickValues
+			, charts
+			, context
+			, xScale2, yScale2
+			, yInsulinScale, yCarbsScale
+			, xAxis2, yAxis2
+			, dateFn = function (d) { return new Date(d.date) }
+			, foodtexts = 0;
+
+        // Tick Values
+		if (options.scale == SCALE_LOG) {
+			if (browserSettings.units == 'mmol') {
+				tickValues = [
+					  2.0
+					, 3.0
+					, options.targetLow
+					, 6.0
+					, 8.0
+					, options.targetHigh
+					, 16.0
+					, 22.0
+				];
+			} else {
+				tickValues = [
+					  40
+					, 60
+					, options.targetLow
+					, 120
+					, 160
+					, options.targetHigh
+					, 250
+					, 400
+				];
+			}
+		} else {
+			if (browserSettings.units == 'mmol') {
+				tickValues = [
+					  2.0
+					, 4.0
+					, 6.0
+					, 8.0
+					, 10.0
+					, 12.0
+					, 14.0
+					, 16.0
+					, 18.0
+					, 20.0
+					, 22.0
+				];
+			} else {
+				tickValues = [
+					  40
+					, 80
+					, 120
+					, 160
+					, 200
+					, 240
+					, 280
+					, 320
+					, 360
+					, 400
+				];
+			}
+		}
+
+        // create svg and g to contain the chart contents
+        charts = d3.select('#'+containerprefix+day).html(
+			'<b>'+
+			localeDate(day)+
+			'</b><br>'
+		).append('svg');
+
+		charts.append("rect")
+			.attr("width", "100%")
+			.attr("height", "100%")
+			.attr("fill", "WhiteSmoke");
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769498838/issues/source?bid=2452374&fileBranchId=2459081#l153)